### PR TITLE
Minor fixes to deployment README.md based on latest tryout, moved notebook to samples folder, more documentation cleanup.

### PR DESCRIPTION
**Pull request by jainr**
Minor Updates
- Moved notebooks to new location under samples
- Product recommendation advance notebook had some stale code, updated it and tested it on databricks.
- Found out that in addition to KV and Storage permission, we also need to add notebook user as Synapse contributor so they can submit jobs. Added that.
- Deploy button was embedded in all notebooks, removed that and asked user to follow the deployment guide.
- Added steps to verify deployment was successful and pointer to run notebook sample.

### CODE REVIEW

**Comment by blrchen** [comment]
>         "# Feathr Feature Store on Azure Demo Notebook\n",

Should this sample get moved as well?

---
Refers to: feathr_project/feathrcli/data/feathr_user_workspace/nyc_driver_demo.ipynb:7 in 3a5285f. [](commit_id = 3a5285f0db3f7abe4aeaa089001ff8f90acf1d6c, deletion_comment = False)

**Comment by blrchen** [comment]
Thanks for the improvements!

Could you please update title to make it more meaningful, and it seems link check does not run as part of PR checks, please watch link check result on main branch to ensure no link breaks.

File: docs/how-to-guides/azure-deployment-arm.md [hunk 1]
```diff
@@ -69,11 +69,20 @@ Feathr has native cloud integration and getting started with Feathr is very stra
...
+3. If you are using Purview registry there is an additional step required for the deployment to work. Registry Server authenticates with Azure Purview using Managed Identity that was created by ARM template. The Managed Identity needs to be added to Azure Purview Collections as a __Data Curator__. For more details, please refer to [Access control in the Microsoft Purview governance portal](https://docs.microsoft.com/en-us/azure/purview/catalog-permissions)
     ![purview data curator role add](../images/purview_permission_setting.png)
 
     Only collection admins can perform the above operation, the user who created this Purview account is already one. If you want to add additional admins, you can do so by clicking on _Root collection permission_ option on Azure Purview page.
 
+Congratulations, you have successfully deployed Feathr on Azure. You can access your resources by going to the resource group that you created for the deployment. A good first test would be to access Feathr UI, you can access it by clicking on App Service URL.
+
+![app service url](../images/app-service-url.png)
```
Comment (Yuqing-cat) [inline docs/how-to-guides/azure-deployment-arm.md:79]: This image also has personal login in the upper right corner.

**Comment by jainr** [comment]
> > ```
> >     "# Feathr Feature Store on Azure Demo Notebook\n",
> > ```
> 
> Should this sample get moved as well?
> 
> Refers to: feathr_project/feathrcli/data/feathr_user_workspace/nyc_driver_demo.ipynb:7 in [3a5285f](https://github.com/linkedin/feathr/commit/3a5285f0db3f7abe4aeaa089001ff8f90acf1d6c). [](commit_id = [3a5285f](https://github.com/linkedin/feathr/commit/3a5285f0db3f7abe4aeaa089001ff8f90acf1d6c), deletion_comment = False)

That specific notebook (nyc_taxi_demo) is referenced in many places and I didn't wanted to move it so last minute and miss any references. The product reco notebook are more hollistic and are only referenced in few places, hence made the switch now for them.

File: docs/how-to-guides/azure-deployment-arm.md (hunk 1 above)
Comment (jainr) [reply docs/how-to-guides/azure-deployment-arm.md:79]: Fixed, thanks for catching.

**Comment by jainr** [comment]
> Thanks for the improvements!
> 
> Could you please update title to make it more meaningful, and it seems link check does not run as part of PR checks, please watch link check result on main branch to ensure no link breaks.

Title fixed.
Checker works collected 210 items                                                                                                                                                 

docs/README.md ..............................                                                                                                                 [ 13%]
docs/quickstart_databricks.md ....                                                                                                                            [ 14%]
docs/quickstart_synapse.md ..............                                                                                                                     [ 19%]
docs/concepts/feathr-concepts-for-beginners.md .....                                                                                                          [ 21%]
docs/concepts/feature-definition.md ......                                                                                                                    [ 24%]
docs/concepts/feature-generation.md ........                                                                                                                  [ 28%]
docs/concepts/feature-join.md ...                                                                                                                             [ 29%]
docs/concepts/point-in-time-join.md .                                                                                                                         [ 30%]
docs/design_docs/feature_registry_abstraction.md ....                                                                                                         [ 31%]
docs/dev_guide/cloud_integration_testing.md ..                                                                                                                [ 32%]
docs/dev_guide/cloud_resource_provision.md ...........                                                                                                        [ 37%]
docs/dev_guide/doc_guide.md ..                                                                                                                                [ 38%]
docs/dev_guide/feathr_overall_release_guide.md .....                                                                                                          [ 40%]
docs/dev_guide/publish_to_maven.md ....                                                                                                                       [ 42%]
docs/dev_guide/python_dev_guide.md ....                                                                                                                       [ 44%]
docs/dev_guide/python_package_release.md .                                                                                                                    [ 45%]
docs/dev_guide/scala_dev_guide.md ......                                                                                                                      [ 47%]
docs/dev_guide/update_python_docs.md ....                                                                                                                     [ 49%]
docs/how-to-guides/azure-deployment-arm.md ................                                                                                                   [ 57%]
docs/how-to-guides/azure-deployment-cli.md ...........                                                                                                        [ 61%]
docs/how-to-guides/build-and-push-feathr-registry-docker-image.md .                                                                                           [ 62%]
docs/how-to-guides/deploy-feathr-api-as-webapp.md .......                                                                                                     [ 65%]
docs/how-to-guides/feathr-configuration-and-env.md ............                                                                                               [ 70%]
docs/how-to-guides/feathr-job-configuration.md ...                                                                                                            [ 70%]
docs/how-to-guides/feathr-udfs.md ........                                                                                                                    [ 74%]
docs/how-to-guides/local-feature-testing.md .                                                                                                                 [ 74%]
docs/how-to-guides/streaming-source-ingestion.md ...                                                                                                          [ 76%]
docs/how-to-guides/troubleshoot-feature-definition.md .                                                                                                       [ 76%]
docs/samples/product_recommendation_demo.ipynb .........                                                                                                      [ 80%]
docs/samples/product_recommendation_demo_advanced.ipynb ..........                                                                                            [ 85%]
docs/samples/customer360/Customer360.ipynb ........                                                                                                           [ 89%]
docs/samples/databricks/databricks_quickstart_nyc_taxi_driver.ipynb ......

======================================================================== 210 passed in 2.20s ========================================================================

File: docs/how-to-guides/azure-deployment-arm.md [hunk 2]
```diff
@@ -69,11 +69,20 @@ Feathr has native cloud integration and getting started with Feathr is very stra
...
+3. If you are using Purview registry there is an additional step required for the deployment to work. Registry Server authenticates with Azure Purview using Managed Identity that was created by ARM template. The Managed Identity needs to be added to Azure Purview Collections as a __Data Curator__. For more details, please refer to [Access control in the Microsoft Purview governance portal](https://docs.microsoft.com/en-us/azure/purview/catalog-permissions)
     ![purview data curator role add](../images/purview_permission_setting.png)
 
     Only collection admins can perform the above operation, the user who created this Purview account is already one. If you want to add additional admins, you can do so by clicking on _Root collection permission_ option on Azure Purview page.
 
+Congratulations, you have successfully deployed Feathr on Azure. You can access your resources by going to the resource group that you created for the deployment. A good first test would be to access Feathr UI, you can access it by clicking on App Service URL.
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/azure-deployment-arm.md:77]: Should we add a link to the provisioned UI?

**Comment by xiaoyongzhu** [comment]
This looks good to me. Have a minor comment where we can just add a link in the doc directly for the Feathr UI, but other than that I'm good.

File: docs/how-to-guides/azure-deployment-arm.md (hunk 2 above)
Comment (jainr) [reply docs/how-to-guides/azure-deployment-arm.md]: Good point, added it

**Review by xiaoyongzhu — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Pull request merged**